packages/chat-headless-react/src/ChatHeadlessProvider.tsx

```diff
@@ -25,27 +29,59 @@ export function ChatHeadlessProvider(
   props: ChatHeadlessProviderProps
 ): JSX.Element {
   const { children, config } = props;
-  // deferLoad is used with sessionStorage so that the children won't be
-  // immediately rendered and trigger the "load initial message" flow before
-  // the state can be loaded from session.
-  const [deferLoad, setDeferLoad] = useState(config.saveToSessionStorage);
 
   const headless = useMemo(() => {
     const configWithoutSession = { ...config, saveToSessionStorage: false };
-    const headless = new ChatHeadless(updateClientSdk(configWithoutSession));
+    const headless = provideChatHeadless(updateClientSdk(configWithoutSession));
     return headless;
   }, [config]);
 
+  return (
+    <ChatHeadlessInstanceProvider
+      saveToSessionStorage={config.saveToSessionStorage}
+      headless={headless}
+    >
+      {children}
+    </ChatHeadlessInstanceProvider>
+  );
+}
+
+/**
+ * Props for {@link ChatHeadlessInstanceProvider}
+ *
+ * @public
+ */
+export type ChatHeadlessInstanceProviderProps = PropsWithChildren<{
+  saveToSessionStorage?: boolean;
```

Comment:
seems odd to me that this is a prop when user would already specified that when they create the headless instance